clang/lib/Driver/Driver.cpp

```diff
@@ -1271,6 +1271,12 @@ Compilation *Driver::BuildCompilation(ArrayRef<const char *> ArgList) {
     if (VFS->setCurrentWorkingDirectory(WD->getValue()))
       Diag(diag::err_drv_unable_to_set_working_directory) << WD->getValue();
 
+  // Check for missing include directories
+  for (auto IncludeDir : Args.getAllArgValues(options::OPT_I_Group)) {
+    if (!llvm::sys::fs::is_directory(IncludeDir))
```

Comment:
I think we should go through the VFS here, something like `VFS.exist(IncludeDir)`